src/otx/core/data/factory.py

```diff
@@ -0,0 +1,57 @@
+"""Factory classes for dataset and transforms."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from otx.core.types.task import OTXTaskType
+from otx.core.types.transformer_libs import TransformLibType
+
+from .dataset.base import OTXDataset, Transforms
+
+if TYPE_CHECKING:
+    from datumaro import DatasetSubset
+
+    from otx.core.config.data import SubsetConfig
+
+__all__ = ["TransformLibFactory", "OTXDatasetFactory"]
+
+
+class TransformLibFactory:
+    """Factory class for transform."""
+
+    @classmethod
+    def generate(cls: type[TransformLibFactory], config: SubsetConfig) -> Transforms:
+        """Create transforms from factory."""
+        if config.transform_lib_type == TransformLibType.MMCV:
+            from .transform_libs.mmcv import MMCVTransformLib
+
+            return MMCVTransformLib.generate(config)
+
+        if config.transform_lib_type == TransformLibType.MMDET:
+            from .transform_libs.mmdet import MMDetTransformLib
+
+            return MMDetTransformLib.generate(config)
+
+        raise NotImplementedError(config.transform_lib_type)
+
+
+class OTXDatasetFactory:
```

Comment:
Why is OTXDatasetFactory implemented as a class? From my perspective, it could be simplified to a function. The same functionality could be achieved with a function definition like `def create_otx_dataset(task, datumaro_ds, config)`.

Reply:
I wanted to give a capability to extend in the future such as
```python
class OTXDatasetFactory:
     def create_from_custom_dataset(task, custom_dataset, config):
           ....
```